ngraph/core/src/op/non_max_suppression.cpp

```
@@ -771,11 +761,213 @@ int64_t op::v5::NonMaxSuppression::max_boxes_output_from_input() const
+
+            float y1 = y_center - height / 2.0;
+            float x1 = x_center - width / 2.0;
+            float y2 = y_center + height / 2.0;
+            float x2 = x_center + width / 2.0;
+
+            current_box[0] = y1;
+            current_box[1] = x1;
+            current_box[2] = y2;
+            current_box[3] = x2;
+        }
+    }
+
+    void normalize_box_encoding(float* boxes,
+                                const Shape& boxes_shape,
+                                const V5BoxEncoding box_encoding)
+    {
+        if (box_encoding == V5BoxEncoding::CORNER)
+        {
+            normalize_corner(boxes, boxes_shape);
+        }
+        else
+        {
+            normalize_center(boxes, boxes_shape);
+        }
+    }
+
+    std::vector<float> get_floats(const HostTensorPtr& input, const Shape& shape)
+    {
+        size_t input_size = shape_size(shape);
+        std::vector<float> result(input_size);
+
+        switch (input->get_element_type())
+        {
+        case element::Type_t::bf16:
+        {
+            bfloat16* p = input->get_data_ptr<bfloat16>();
+            for (size_t i = 0; i < input_size; ++i)
+            {
+                result[i] = float(p[i]);
+            }
+        }
+        break;
+        case element::Type_t::f16:
+        {
+            float16* p = input->get_data_ptr<float16>();
+            for (size_t i = 0; i < input_size; ++i)
+            {
+                result[i] = float(p[i]);
+            }
+        }
+        break;
+        case element::Type_t::f32:
+        {
+            float* p = input->get_data_ptr<float>();
+            memcpy(result.data(), p, input_size * sizeof(float));
+        }
+        break;
+        case element::Type_t::f64:
+        {
+            double* p = input->get_data_ptr<double>();
+            for (size_t i = 0; i < input_size; ++i)
+            {
+                result[i] = float(p[i]);
+            }
+        }
+        break;
+        default:;
```

For other precision you returns vector with the garbage (you don't fill this vector), it is not correct.

---

Other precisions are not used due to `void op::v5::NonMaxSuppression::validate()`.

---

In is not a reason to have this unexpected behaviour.

Please resolve this issue. At least (but it is not the best) you can throw an exception for this unsupported case.

---

Fixed.